### PR TITLE
[AUT-2484] Add 'How to review' section in PR template

### PR DESCRIPTION
**Pull request by whi-tw**
### Added How to review section

This is a space for developers to list specific steps needed for the PR to be
fully reviewed. 'Code review' is always implied, but can be stated here.

### Merge 'What' and 'Why' section

This seems completely unnecessary - it makes it really difficult to write
the PR in a logical way: 'what?' and 'why?' are intrinsicly linked, and the description
is easier to understand if these are together, rather than split apart.

### Convert instructions to comments

This emphasises their 'differentness', and means that they are not rendered if
left in after submitting the PR. Some people find it easier to write with the prompt
visible at the time, and this means they don't need to worry about removing the prompt
after finishing writing the description.

## How to review

1. Ensure no important sections have been removed
2. Ensure no unnecessary sections have been added
3. Ensure the instructions are understandable and correct

## Related PRs

https://github.com/alphagov/di-infrastructure/pull/577
https://github.com/govuk-one-login/authentication-check-experian/pull/48
https://github.com/govuk-one-login/authentication-frontend/pull/1423
https://github.com/govuk-one-login/authentication-api/pull/4045

### CODE REVIEW

**Comment by gtvj** [comment]
Thanks @whi-tw - I think these changes make sense but will hold off on approving so others can comment 👍🏽

**Review by dbes-gds — APPROVED** [review]

**Review by BeckaL — APPROVED** [review]
Looks like sensible set of changes to me :-)

**Review by BeckaL — APPROVED** [review]

**Pull request merged**
